lib/async/http/faraday/adapter.rb

```diff
@@ -48,8 +50,41 @@ def read
 				end
 			end
 			
+			class ParallelManager
+				def initialize(options = {})
+					@options = options
+					@barrier = nil
+				end
+				
+				def run
+					raise NotImplementedError, "Please update your Faraday version!"
```

Comment:
nit: This will fail AFTER the old Faraday version already `yield` and therefore after the requests have been fired. Would it make more sense to move this check in the initialiser for an earlier guard?

Reply:
Alternatively, there's really no reason to `raise` here.
The method could simply no-op and issue a warning, explaining that although the requests were performed, they were not executed in parallel because of the old Faraday version

Reply:
I think issuing a warning would be a good idea, you are right, it's a degenerate case, but in theory it should still work.